indico/modules/events/papers/controllers/display.py

```diff
@@ -37,7 +37,8 @@ def _check_paper_protection(self):
 
     def _check_access(self):
         RHPaperBase._check_access(self)
-        if not self.event.cfp.is_open:
+        cfp = self.event.cfp
+        if not cfp.is_open and not cfp.is_authorized_submitter(session.user):
```

Comment:
Would it make sense to have a method in the cfp object that checks both? I see two cases here and at least one in the templates where we want to know if the user can submit papers (regardless of the reason).

Reply:
Good point, added `can_submit_proceedings` to CfP